path_rotate.go

```diff
@@ -159,9 +161,16 @@ func (b *backend) pathRotateRoleCredentialsUpdate(ctx context.Context, req *logi
 		return nil, pushErr
 	}
 
+	if err != nil {
+		return &logical.Response{
```

Comment:
I haven't tested yet, but I think this combination of non-error response object, plus an error will return a 400 bad request status code from the API, which isn't ideal.

Reply:
Perhaps it would be better to just return what is currently a warning string as an error, and include the underlying error in that string?

Reply:
Good point, yeah including it in the error message sounds good!